flask_static_digest/__init__.py

```diff
@@ -26,28 +27,30 @@ def init_app(self, app):
         app.config.setdefault("FLASK_STATIC_DIGEST_HOST_URL", None)
 
         self.host_url = app.config.get("FLASK_STATIC_DIGEST_HOST_URL")
-        self.static_url_path = app.static_url_path
 
-        self.manifest_path = os.path.join(app.static_folder,
-                                          "cache_manifest.json")
-        self.has_manifest = os.path.exists(self.manifest_path)
+        self.manifests = {}
 
-        self.manifest = {}
-
-        if self.has_manifest:
-            self.manifest = self._load_manifest(app)
+        self._load_manifest("static", app)
+        for endpoint, blueprint in app.blueprints.items():
+            self._load_manifest(f"{endpoint}.static", blueprint)
 
         app.add_template_global(self.static_url_for)
 
-    def _load_manifest(self, app):
-        with app.open_resource(self.manifest_path, "r") as f:
-            manifest_dict = json.load(f)
-
-        return manifest_dict
-
-    def _prepend_host_url(self, host, filename):
-        return urljoin(self.host_url,
-                       "/".join([self.static_url_path, filename]))
+    def _load_manifest(self, endpoint, scaffold):
+        if not scaffold.has_static_folder:
+            return
+
+        manifest_path = os.path.join(scaffold._static_folder,
+                                     "cache_manifest.json")
+        try:
```

Comment:
Nice touch on the error detection, since this only loads once when the extension loads it's worth it.